src/comments.js

```diff
@@ -823,11 +830,39 @@ function handleVariableDeclaratorComments(
   return false;
 }
 
+function handleTrailingGraphQLComments(precedingNode, comment) {
+  if (
+    precedingNode &&
+    precedingNode.kind &&
+    (precedingNode.kind === "Field" ||
+      precedingNode.kind === "VariableDefinition")
+  ) {
+    addLeadingComment(precedingNode, comment);
+    return true;
+  }
+  return false;
+}
+
+function handleLeadingGraphQLComments(followingNode, comment) {
+  if (
+    followingNode &&
+    followingNode.kind &&
+    (followingNode.kind === "Field" ||
+      followingNode.kind === "OperationDefinition")
+  ) {
+    addLeadingComment(followingNode, comment);
+    return true;
+  }
+  return false;
+}
```

Comment:
This is weird, the generic logic checks if the comment is on its own line and keeps it that way by adding a \n before. Maybe the comment location is off by one and when going backwards it sees the # characters and says it's not on its own line.